### PR TITLE
[BUG] [Singleton Service] new category "dateFields" (relative pattern data) doesn't support in combination API 

### PR DESCRIPTION
**Pull request by chaoq0318**
fix issue:https://github.com/vmware/singleton/issues/299

### CODE REVIEW

File: g11n-ws/modules/md-restful-api-i18n/src/main/java/com/vmware/vip/i18n/api/base/utils/CommonUtility.java [hunk 1]
```diff
@@ -5,35 +5,39 @@
...
+     * @param scope
+     * @param perfectMatchFlag
+     * If perfectMatchFlag = true and not found from the CategoriesEnum, return null
+     * If perfectMatchFlag = false and not found from the CategoriesEnum，return the original value
+     *
      * @return
      */
-    public static boolean checkParams(List<String> categories, String... args){
-        List<String> newCategories = new ArrayList<>();
+    public static List<String> getCategoriesByEnum(String scope, boolean perfectMatchFlag){
```
Comment (huihuiw01) [inline g11n-ws/modules/md-restful-api-i18n/src/main/java/com/vmware/vip/i18n/api/base/utils/CommonUtility.java]: I think name the parameter as 'perfectMatch' is enough.

Comment (chaoq0318) [reply g11n-ws/modules/md-restful-api-i18n/src/main/java/com/vmware/vip/i18n/api/base/utils/CommonUtility.java]: fixed

File: g11n-ws/modules/md-restful-api-i18n/src/main/java/com/vmware/vip/i18n/api/base/utils/CommonUtility.java [hunk 2]
```diff
@@ -5,35 +5,39 @@
...
-        }
-        categories.clear();
-        categories.addAll(newCategories);
-        for (String param : args){
-            if (CommonUtil.isEmpty(param)){
-                return false;
+                if (perfectMatchFlag) {
+                    return null;
+                }
+                categoryList.add(cat);
```
[the remark this answers is not in the document]
Comment (chaoq0318) [reply g11n-ws/modules/md-restful-api-i18n/src/main/java/com/vmware/vip/i18n/api/base/utils/CommonUtility.java]: For the combine API, when a category does not match, other matching categories can also be queried
When perfectMatch = false, the original category will be added

File: g11n-ws/modules/md-restful-api-i18n/src/main/java/com/vmware/vip/i18n/api/base/TranslationWithPatternAction.java [hunk 3]
```diff
@@ -131,16 +132,15 @@ public boolean isPatternNoRegion(TranslationWithPatternDTO data) {
 	   @SuppressWarnings("unchecked")
 	   public Map<String, Object> getPattern(TranslationWithPatternDTO data) throws Exception {
 	      Map<String, Object> pattern = null;
+		   List<String> categories = CommonUtility.getCategoriesByEnum(data.getScope(), false);
```
[the remark this answers is not in the document]
Comment (chaoq0318) [reply g11n-ws/modules/md-restful-api-i18n/src/main/java/com/vmware/vip/i18n/api/base/TranslationWithPatternAction.java]: For the combine API, when a category does not match, other matching categories can also be queried
When perfectMatch = false, the original category will be added

**Pull request merged**